src/main/webapp/script.js

```diff
@@ -223,29 +229,43 @@ function createRecipeElement(recipe) {
 
 /** Get profile information to determine which alerts to create */
 function createRecipeCardAlerts(recipe, alertElements) {
-  alertElements.appendChild(createAlertElement("icon-warning-sign", "Dietary Alert"));
-  alertElements.appendChild(createAlertElement("icon-exclamation", "Dietary Alert"));
-  alertElements.appendChild(createAlertElement("icon-leaf", "Non-Vegetarian Alert"));
-  alertElements.appendChild(createAlertElement("icon-coffee", "Non-DairyFree Alert"));
-  alertElements.appendChild(createAlertElement("icon-food", "Allergies Alert"));
-  //   //fetch('/profile').then(response => response.json()).then((message) => {
-
-  //     if (message.hasProfile) {
-  //       const profile = message.profile;
-  //       // TODO: need comparisons against recipe json
-  //       // profile.vegetarian
-  //       // profile.vegan
-  //       // profile.glutenFree
-  //       // profile.dairyFree
-  //       // profile.allergies
-
-  //       alertElements.appendChild(createAlertElement("icon-warning-sign", "Dietary Alert"));
-  //       alertElements.appendChild(createAlertElement("icon-exclamation", "Dietary Alert"));
-  //       alertElements.appendChild(createAlertElement("icon-leaf", "Non-Vegetarian Alert"));
-  //       alertElements.appendChild(createAlertElement("icon-coffee", "Non-DairyFree Alert"));
-  //       alertElements.appendChild(createAlertElement("icon-food", "Allergies Alert"));
-  //     }
-  //   });
+  fetch('/profile').then(response => response.json()).then((message) => {
+
+    if (message.hasProfile) {
+      const profile = message.profile;
+      if (profile.vegetarian && !recipe["vegetarian"]) {
+        alertElements.appendChild(createAlertElement("icon-leaf", "Non-Vegetarian Alert"));
+      }
+      if (profile.vegan && !recipe["vegan"]) {
+        alertElements.appendChild(createAlertElement("icon-exclamation", "Non-Vegan Alert"));
+      }
+      if (profile.glutenFree && !recipe["glutenFree"]) {
+        alertElements.appendChild(createAlertElement("icon-warning-sign", "Non-GlutenFree Alert"));
+      }
+      if (profile.dairyFree && !recipe["dairyFree"]) {
+        alertElements.appendChild(createAlertElement("icon-coffee", "Non-DairyFree Alert"));
+      }
```

Comment:
I think these could be refactored into a loop over tag values and map lookup:

for (tag of tags) {
  if (profile[tag] && !recipe[tag]) {
    alertElements.appendChild(createAlertElement(icons[tag], messages[tag]));
  }
}